src/traces/contourcarpet/index.js

```diff
@@ -15,7 +15,7 @@ ContourCarpet.supplyDefaults = require('./defaults');
 ContourCarpet.colorbar = require('../contour/colorbar');
 ContourCarpet.calc = require('./calc');
 ContourCarpet.plot = require('./plot');
-ContourCarpet.style = require('./style');
```

Comment:
hmm, as far as I can tell only `cheater_contour` is *not* constraint-type, but perhaps I can extend that one to cover these cases. Sure.